### PR TITLE
Fix mistakes in pull #3864

### PR DESCRIPTION
**Pull request by Infiltrator**
Bug unittests should go in their own blocks.

### CODE REVIEW

File: std/algorithm/sorting.d [hunk 1]
```diff
@@ -2162,7 +2162,7 @@ auto topN(alias less = "a < b",
 
     static assert(ss == SwapStrategy.unstable,
             "Stable topN not yet implemented");
-    auto ret = r[0 .. nth];
+    auto ret = r[0 .. nth].save;
```
Comment (JakobOvrum) [inline std/algorithm/sorting.d]: I don't think `save` makes any sense here

**Comment by JakobOvrum** [comment]
> ranges should be .saved.

It makes no sense when it's on a slice result. The slice is already a distinct iterator, otherwise the type would be not be able to implement slicing.

**Comment by Infiltrator** [comment]
Good point.

**Comment by Infiltrator** [comment]
That's why I keep screwing up the unittest in the autobuilder.  I forgot the std/algorithm.test after make.

**Comment by JackStouffer** [comment]
LGTM

File: std/algorithm/sorting.d [hunk 2]
```diff
@@ -2317,11 +2321,13 @@ unittest
...
     sort(a);
     assert(a == [0, 1, 2, 2, 3]);
+}
 
-    // bug 12987
-    int[] c = [ 5, 7, 2, 6, 7 ];
-    int[] d = [ 2, 1, 5, 6, 7, 3, 0 ];
-    auto t = topN(c, d);
+// bug 12987
+unittest {
```
Comment (quickfur) [inline std/algorithm/sorting.d]: Nitpick: opening brace on its own line please.

**Comment by quickfur** [comment]
Other than that, LGTM.

**Comment by Infiltrator** [comment]
Fixed the brace; and hopefully my config for future phobos bracing.

**Comment by JakobOvrum** [comment]
Auto-merge toggled on

**Pull request merged**